new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItServerStartPolicy.java

```diff
@@ -0,0 +1,1237 @@
+        .append(" \"value\":  \"IF_NEEDED\"")
+        .append(" }]");
+    logger.info("Cluster restart patch string: {0}", patchStr);
+
+    patch = new V1Patch(new String(patchStr));
+    crdPatched = false;
+    crdPatched = assertDoesNotThrow(() ->
+            patchDomainCustomResource(domainUid, domainNamespace, patch, "application/json-patch+json"),
+        "patchDomainCustomResource(clusterRestart) failed");
+
+    assertTrue(crdPatched, "patchDomainCustomResource(clusterRestart) failed");
+    logger.info("Wait for Cluster server pod {0} to be ready in namespace {1}",
+        configServerPodName, domainNamespace);
+    checkPodReady(configServerPodName, domainUid, domainNamespace);
+
+    logger.info("Check configured cluster managed service {0} is created in namespace {1}",
+        configServerPodName, domainNamespace);
+    checkServiceExists(configServerPodName, domainNamespace);
+    logger.info("Configured cluster restart success");
+  }
+
+  /**
+   * Stop a dynamic cluster by patching the resource defintion with 
+   *  spec/clusters/1/serverStartPolicy set to NEVER.
+   * Make sure that ONLY servers in second Cluster are stopped. 
+   * Restart the dynamic cluster by patching the resource defintion with 
+   *  spec/clusters/1/serverStartPolicy set to IF_NEEDED.
+   * Make sure that servers in second cluster are in RUNNING state. 
+   */
+  @Test
+  @DisplayName("Restart the dynamic cluster with serverStartPolicy")
+  public void testDynamicClusterRestart() {
+
+    String dynamicServerPodName = domainUid + "-managed-server1";
+    String configServerPodName = domainUid + "-config-cluster-server1";
+
+    // Make sure that dynamic cluster managed server is Ready 
+    checkPodReady(dynamicServerPodName, domainUid, domainNamespace);
+    checkServiceExists(dynamicServerPodName, domainNamespace);
+    logger.info("(BeforePatch) dynamic cluster Managed Server is RUNNING");
+
+    StringBuffer patchStr = null;
+    patchStr = new StringBuffer("[{");
+    patchStr.append("\"op\": \"replace\",")
+        .append(" \"path\": \"/spec/clusters/0/serverStartPolicy\",")
+        .append(" \"value\":  \"NEVER\"")
+        .append(" }]");
+    logger.info("Dynamic cluster shutdown patch string: {0}", patchStr);
+
+    patch = new V1Patch(new String(patchStr));
+    boolean crdPatched = assertDoesNotThrow(() ->
+            patchDomainCustomResource(domainUid, domainNamespace, patch, "application/json-patch+json"),
+        "patchDomainCustomResource(dynamicClusterShutdown) failed");
+    assertTrue(crdPatched, "patchDomainCustomResource(dynamicClusterShutdown) failed");
+   
+    checkPodDeleted(dynamicServerPodName, domainUid, domainNamespace);
+    logger.info("Dynamic cluster shutdown success");
+
+    // check managed server from other cluster are not affected
+    logger.info("Check managed server pods are not affected");
+    checkPodReady(configServerPodName, domainUid, domainNamespace);
```

Comment:
server pod is not effected

Reply:
Here we make sure all the managed server(s) pod from other cluster are not affected.

Reply:
You have to save the pod creation timestamps before you attempt to restart servers and then after intended server is restarted verify other pods creation timestamps is not changed.

Reply:
Modified the logic to compare the pod creation timestamp